### PR TITLE
LR1 init

### PR DESCRIPTION
**Pull request by Sergey-Dvoraninovich**

### CODE REVIEW

File: LR1/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    defaultConfig {
+        applicationId "com.example.myapplication"
+        minSdkVersion 28
+        targetSdkVersion 30
+        versionCode 1
+        versionName "1.0"
```
Comment (gmltA) [inline LR1/app/build.gradle]: App versioning is not implemented.

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                adapter_down.setDropDownViewResource(layout.simple_spinner_dropdown_item);
+                spinner_down.setAdapter(adapter_down);
+                SaveDownViewModel model_order_down = new ViewModelProvider(requireActivity()).get(SaveDownViewModel.class);
+                model_order_down.getSelected().observe(getViewLifecycleOwner(), new Observer<Integer>() {
+                    @Override
+                    public void onChanged(Integer order) {
+                        spinner_down.setSelection(order);
+                    }
+                });
+            }
```
Comment (gmltA) [inline LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: Can you extract similar code from observers?

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view,
+                                       int position, long id) {
+                new_setConv(null);
+                SaveDownViewModel model_order = new ViewModelProvider(requireActivity()).get(SaveDownViewModel.class);
+                model_order.select(position);
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+            }
```
Comment (gmltA) [inline LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: ... and from the spinner listeners?

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+                        ans_double /= 1000000;
+                        ans_text = Double.toString(ans_double);
+
+                    } catch (Exception e) {
+                        ans_text = "";
+                    }
+                }
+                text_down.setText(ans_text);
+                SaveViewModel model_order = new ViewModelProvider(requireActivity()).get(SaveViewModel.class);
+                model_order.select(new_line_value);
```
Comment (gmltA) [inline LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: All the conversion logic have to be extracted from the fragment (View level) to separate class. ViewModel should be responsible for conversion via this new class. Fragments should only display data from the view model.

File: LR1/app/src/main/java/com/example/myapplication/BlankFragment.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link BlankFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class BlankFragment extends Fragment {
```
Comment (gmltA) [inline LR1/app/src/main/java/com/example/myapplication/BlankFragment.java:16]: What is this thing for?

File: LR1/app/src/main/java/com/example/myapplication/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener{
+
+    private Conversion conv;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Button button_weight = (Button)findViewById(R.id.button_weignt);
```
Comment (gmltA) [inline LR1/app/src/main/java/com/example/myapplication/MainActivity.java:24]: You should not access fragment's view from the activity. Move this logic to the fragment. You can still access the same view model from the fragment.

File: LR1/app/src/main/java/com/example/myapplication/SaveTopViewModel.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.example.myapplication;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+public class SaveTopViewModel extends ViewModel {
```
Comment (gmltA) [inline LR1/app/src/main/java/com/example/myapplication/SaveTopViewModel.java:7]: You have a separate ViewModel for each value, am I right?

**Review by gmltA — CHANGES_REQUESTED** [review]

File: LR1/app/build.gradle (hunk 1 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/build.gradle]: Добавлено

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java (hunk 2 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: Изменено

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java (hunk 3 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: Изменено

File: LR1/app/src/free/java/com/example/myapplication/WeightFragment.java (hunk 4 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/free/java/com/example/myapplication/WeightFragment.java]: Изменено

File: LR1/app/src/main/java/com/example/myapplication/BlankFragment.java (hunk 5 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/main/java/com/example/myapplication/BlankFragment.java:16]: Не используется в проекте. Удалено в новом коммите

File: LR1/app/src/main/java/com/example/myapplication/MainActivity.java (hunk 6 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/main/java/com/example/myapplication/MainActivity.java:24]: Я не использовал фрагмент в этом активити

File: LR1/app/src/main/java/com/example/myapplication/SaveTopViewModel.java (hunk 7 above)
Comment (Sergey-Dvoraninovich) [reply LR1/app/src/main/java/com/example/myapplication/SaveTopViewModel.java:7]: Раньше было так. Исправлено в новой версии

**Review by gmltA — APPROVED** [review]
Now it's better.

**Pull request merged**